forte/data/data_store.py

```diff
@@ -174,72 +179,109 @@ def add_annotation_raw(
 
-    def set_attr(self, tid: int, attr_id: int, attr_value):
+    def set_attr(self, tid: int, attr_id: int, attr_value: Any):
         r"""This function locates the entry data with `tid` and sets its
-        attribute `attr_id` with value `attr_value`.
+        attribute `attr_id` with value `attr_value`. Called by `set_attribute()`.
 
         Args:
             tid (int): Unique id of the entry.
-            attr_id (int): id of the attribute.
-            attr_value: value of the attribute.
-
+            attr_id (int): Id of the attribute.
+            attr_value (any): Value of the attribute.
         """
-        # We retrieve the entry data from entry_dict using tid. We get its
-        # entry type. We then locate the attribute using attribute if, and update the attribute.
-
-        raise NotImplementedError
+        # We retrieve the entry data from `entry_dict` using tid.
+        # We locate the attribute using `attr_id` and update the attribute.
+        entry = self.entry_dict[tid]
+        entry[attr_id] = attr_value
 
     def get_attr(self, tid: int, attr_id: int):
         r"""This function locates the entry data with `tid` and gets the value
-        of `attr_id` of this entry.
+        of `attr_id` of this entry. Called by `get_attribute()`.
 
         Args:
             tid (int): Unique id of the entry.
-            attr_id (int): id of the attribute.
+            attr_id (int): Id of the attribute.
 
         Returns:
             The value of `attr_id` for the entry with `tid`.
         """
-        # We retrieve the entry data from entry_dict using tid. We get its
-        # entry type. We then locate the attribute using attr_id, and get the attribute.
+        # We retrieve the entry data from `entry_dict` using tid.
+        # We locate the attribute using `attr_id` and get the attribute.
+        entry = self.entry_dict[tid]
+        return entry[attr_id]
 
-        raise NotImplementedError
+    def set_attribute(self, tid: int, attr_name: str, attr_value: Any):
+        r"""This function locates the entry data with `tid` and finds `attr_id`
+        of its attribute `attr_name`. `Tid`, `attr_id`, and `attr_value` are
+        passed to `set_attr()`.
+
+        Args:
+            tid (int): Unique Id of the entry.
+            attr_name (str): Name of the attribute.
+            attr_value (any): Value of the attribute.
+        """
+        if tid not in self.entry_dict:
+            raise KeyError(f"Entry with tid {tid} not found.")
+        entry_type = self.entry_dict[tid][self.entry_type_idx]
+        if attr_name not in self._type_attributes[entry_type]:
+            raise ValueError(f"{entry_type} has no {attr_name} attribute.")
+        attr_id = self._type_attributes[entry_type][attr_name]
+        self.set_attr(tid, attr_id, attr_value)
+
+    def get_attribute(self, tid: int, attr_name: str):
+        r"""This function locates the entry data with `tid` and finds `attr_id`
```

Comment:
To write the docstring here, should talk about the functionality first, not the implementation detail first.